server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -789,6 +798,55 @@ func extractOptionsFromStartedBuildEvent(event *build_event_stream.BuildEvent) (
 	return "", nil
 }
 
+func getNumActionsShownFromStartedBuildEvent(event *build_event_stream.BuildEvent) int {
+	options, err := extractOptionsFromStartedBuildEvent(event)
+	if err != nil {
+		log.Warningf("Could not extract options for ui_actions_show, defaulting to 8: %v", err)
+		return 8
+	}
+	optionsList, err := shlex.Split(options)
+	if err != nil {
+		log.Warningf("Could not shlex split options for ui_actions_show, defaulting to 8: %v", err)
+		return 8
+	}
+	actionsShownValues := getOptionValues(optionsList, "ui_actions_shown")
+	cursesValues := getOptionValues(optionsList, "curses")
+	if len(cursesValues) > 1 {
+		log.Warningf("Too many arguments to curses, assuming auto: %v", cursesValues)
+	}
+	if len(cursesValues) == 1 && cursesValues[0] == "no" {
+		return 0
+	}
+	if len(actionsShownValues) > 1 {
+		log.Warningf("Too many arguments to ui_actions_shown, defaulting to 8: %v", actionsShownValues)
+	}
+	if len(actionsShownValues) == 1 {
+		n, err := strconv.Atoi(actionsShownValues[0])
+		if err != nil {
+			log.Warningf("Invalid argument to ui_actions_shown, defaulting to 8: %v", err)
+		} else if n < 1 {
+			return 1
+		} else {
+			return n
+		}
+	}
+	return 8
+}
+
+func getOptionValues(options []string, optionName string) []string {
```

Comment:
Instead of collecting all the values, it would be simpler to return the value that appears the latest. That should match the Bazel flag semantics.